src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java

```diff
@@ -1,117 +1,100 @@
-    static final String PDB_REPORT_COLUMNS =
-            "structureId,collectionTemperature,sequence,macromoleculeType,structureMolecularWeight," +
-            "spaceGroup,lengthOfUnitCellLatticeA,lengthOfUnitCellLatticeB,lengthOfUnitCellLatticeC," +
-            "unitCellAngleAlpha,unitCellAngleBeta,unitCellAngleGamma,resolution,name," +
-            "residueCount,atomSiteCount,structureTitle,pdbDoi,structureAuthor,depositionDate," +
-            "releaseDate,revisionDate,experimentalTechnique,title,pubmedId,citationAuthor," +
-            "journalName,publicationYear";
-
-    private final String baseUrl;
-    private final XmlMapper xmlMapper;
+    static final String BASE_ENDPOINT_URL = "https://data.rcsb.org/rest/v1/core";
+    static final String ENTRY_PATH = "entry";
+    static final String POLYMER_ENTITY_PATH = "polymer_entity";
+
+    private String baseUrl;
+    private ObjectMapper objectMapper;
 
     // -------------------- CONSTRUCTORS --------------------
 
     public PdbApiCaller() {
-        this("https://www.rcsb.org");
+        this(BASE_ENDPOINT_URL);
     }
 
     PdbApiCaller(String baseUrl) {
         this.baseUrl = baseUrl;
-        this.xmlMapper = new XmlMapper();
-        this.xmlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        this.objectMapper = new ObjectMapper();
+        this.objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
     }
 
     // -------------------- LOGIC --------------------
 
-    /**
-     * Obtains PDB data for given protein. Also accounts for a fact, that instead of standard HTTP
-     * <code>404 Not Found</code>, the API returns an empty XML to represent unknown structure identifiers.
-     * @param structureId PDB identifier of the structure to get report for.
-     * @return PDB report for given structure if found.
-     */
-    Optional<Dataset> getStructureData(String structureId) {
-        URI reportUri = new SafeUriBuilder(baseUrl).buildWith(builder -> builder
-                .setPath(PDB_REPORT_ENDPOINT)
-                .addParameter(PDB_IDS_PARAM_NAME, structureId)
-                .addParameter(PRIMARY_ONLY_PARAM_NAME, "1")
-                .addParameter(CUSTOM_REPORT_COLUMNS_PARAM_NAME, PDB_REPORT_COLUMNS));
-
-        return ofNullable(executeForObject(Request::Get, reportUri, Dataset.class))
-                .filter(Dataset::hasRecords);
+    public StructureData getStructureData(String structureId) {
+
+        URI entryEndpointUri = buildUri(baseUrl, ENTRY_PATH, structureId);
+        EntryData entryData = getAndMap(entryEndpointUri, EntryData.class);
+        StructureData structureData = new StructureData(entryData);
+
+        Integer polymerEntityCount = structureData.getEntryData()
+                .getRcsbEntryInfo()
+                .getPolymerEntityCount();
+
+        if (polymerEntityCount == null || polymerEntityCount < 1) {
```

Comment:
I'm thinking if we should have some kind of documentation for this kind of checks or is it enough to just learn pdb domain, what do you think?

Reply:
I'll write a comment, but generally I think we do not have consistent strategy for such issues and that's something we can discuss tomorrow on daily.